src/lib/vasi-macro/src/lib.rs

```diff
@@ -8,28 +8,43 @@ use syn::{parse_quote, Attribute, GenericParam, Generics, Type};
 /// Implement `vasi::VirtualAddressSpaceIndependent` for the annotated type.
 /// Requires all fields to implement `vasi::VirtualAddressSpaceIndependent`.
 ///
-/// An empty struct trivially qualifies:
-/// ```
+/// An empty struct fails becase Rust doesn't consider it FFI-safe:
+/// ```compile_fail
 /// use vasi::VirtualAddressSpaceIndependent;
 ///
 /// #[derive(VirtualAddressSpaceIndependent)]
-/// struct Foo {}
+/// #[repr(C)]
+/// struct FooC {}
+///
+/// #[repr(transparent)]
+/// #[derive(VirtualAddressSpaceIndependent)]
+/// struct FooTransparent {}
 /// ```
```

Comment:
Why does this doc comment fail to compile? They're empty structs, one is repr(C) and the other is repr(transparent)? Or are zero-sized types not ffi safe?

Reply:
```
error: `extern` fn uses type `FooC`, which is not FFI-safe
 --> src/lib.rs:17:8
  |
7 | struct FooC {}
  |        ^^^^ not FFI-safe
  |
  = help: consider adding a member to this struct
  = note: this struct has no fields
```

Tweaked and broke up this example a bit for clarity